app/src/main/java/org/projectbuendia/client/diagnostics/BuendiaApiHealthCheck.java

```diff
@@ -158,6 +158,7 @@ public BuendiaModuleHealthCheckRunnable(Handler handler) {
                             case HttpURLConnection.HTTP_FORBIDDEN:
                             case HttpURLConnection.HTTP_UNAUTHORIZED:
                                 reportIssue(HealthIssue.SERVER_AUTHENTICATION_ISSUE);
+                                mHandler.getLooper().quit();
```

Comment:
Nice. It might be better to call `stop()` or `stopImpl()` instead of quitting the handler directly though.